internal/storage/auth/keycloak/user.go

```diff
@@ -83,21 +83,50 @@ func (r *UserRepository) RemoveSession(ctx context.Context, refreshToken string)
 	return nil
 }
 
+func (r *UserRepository) GetAll(ctx context.Context) ([]*entities.User, error) {
+	clientToken, err := loginRestAPIClient(ctx, r.cfg.OidcProvider.BaseURL, r.cfg.OidcProvider.Backend.ClientID, r.cfg.OidcProvider.Backend.ClientSecret, r.cfg.OidcProvider.DomainName)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to log in to Keycloak")
+	}
+	client := gocloak.NewClient(r.cfg.OidcProvider.BaseURL)
+
+	users, err := client.GetUsers(ctx, clientToken.AccessToken, r.cfg.OidcProvider.DomainName, gocloak.GetUsersParams{})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get users from Keycloak")
+	}
+
+	var allUsers []*entities.User
+
+	for _, kcUser := range users {
+		user, err := keyCloakUserToUser(kcUser)
+		if err != nil {
+			continue
```

Comment:
Why should it be continued if an error is happening?

Reply:
Continuing would be similar to skipping users that are nil, we don't need to add them to the response list.
It's the same as saying: `if user != nil { allUsers = append(allUsers, user) }`

But what do you suggest, should we break the loop if the conversion from a Keycloak user to user fails and return an error ?

Reply:
I would say so, yes ...